server/src/controllers/profile/editPassword.ts

```diff
@@ -0,0 +1,18 @@
+import { editPasswordDB, getUserById } from '../../database/queries';
+import { comparePassword, editPasswordSchema, hashPassword } from '../../utils';
+
+const editPassword = async (req, res) => {
+  const { oldPassword, newPassword } = req.body;
+  const id = 1;
+  await editPasswordSchema.validateAsync(req.body);
+  const { rows } = await getUserById(id);
+  const { password } = rows[0];
+  const isMatch = await comparePassword(oldPassword, password);
+  if (!isMatch) {
+    return res.status(400).json({ message: 'wrong password !' });
+  }
+  const hashedNewPassword = await hashPassword(newPassword);
```

Comment:
here, you should check if the old password equal new password, then you will send a response to tell the client side your new password matching your old password.